okio/src/main/java/okio/Buffer.java

```diff
@@ -2092,6 +2092,48 @@ public long resizeBuffer(long newSize) {
       return oldSize;
     }
 
+    /**
+     * Grow the buffer by adding a <strong>contiguous range</strong> of capacity in a single
+     * segment. This adds at least {@code minByteCount} bytes but may add up to a full segment of
+     * additional capacity.
+     *
+     * <p>If {@code minByteCount} bytes are available in the buffer's current tail segment that will
+     * be used; otherwise another segment will be allocated and appended. In either case this
+     * returns the number of bytes of capacity added to this buffer.
+     *
+     * <p>Warning: it its the caller’s responsibility to either write new data to every byte of the
+     * newly-allocated capacity, or to {@link #resizeBuffer shrink} the buffer to the data written.
+     * Failure to do so may cause serious security problems as the data in the returned buffers is
+     * not zero filled. Buffers may contain dirty pooled segments that hold very sensitive data from
+     * other parts of the current process.
+     *
+     * @param minByteCount the size of the contiguous capacity. Must be positive and less than the
+     *     capacity size of a single segment (8 KiB).
+     * @return the number of bytes expanded by. Not less than {@code minByteCount}.
+     */
+    public long expandBuffer(int minByteCount) {
+      if (minByteCount <= 0) {
```

Comment:
It seems odd to check the lower bound here, but not the upper bound.
`Buffer.writableSegment()` checks both, but doesn't provide a helpful error message.

Reply:
Good call. Fixed.